core/services/keeper/registry_synchronizer_process_logs.go

```diff
@@ -0,0 +1,193 @@
+package keeper
+
+import (
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/keeper_registry_wrapper"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/log"
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+func (rs *RegistrySynchronizer) processLogs(head models.Head) {
+	wg := sync.WaitGroup{}
+	wg.Add(4)
+	go rs.handleSyncRegistryLog(head, wg.Done)
+	go rs.handleUpkeepCanceledLogs(head, wg.Done)
+	go rs.handleUpkeepRegisteredLogs(head, wg.Done)
+	go rs.handleUpkeepPerformedLogs(head, wg.Done)
+	wg.Wait()
+}
+
+func (rs *RegistrySynchronizer) handleSyncRegistryLog(head models.Head, done func()) {
+	defer done()
+	oldEnough := isOldEnoughConstructor(head, rs.minConfirmations)
+	i := rs.mailRoom.mbSyncRegistry.RetrieveIf(oldEnough)
+	if i == nil {
+		return
+	}
+	broadcast, ok := i.(log.Broadcast)
+	if !ok {
+		logger.Errorf("RegistrySynchronizer: invariant violation, expected log.Broadcast but got %T", broadcast)
+		return
+	}
+	was, err := broadcast.WasAlreadyConsumed()
+	if err != nil {
+		logger.Warn(err)
+		return
+	}
+	if was {
+		return
+	}
+	if !head.IsInChain(broadcast.RawLog().BlockHash) {
+		return
+	}
+	_, err = rs.syncRegistry()
+	if err != nil {
+		logger.Error(err)
```

Comment:
maybe you also need Errorf here with some more description?

Reply:
:+1: